### PR TITLE
[NEW] Typing +:emoji_name: reacts to last message

### LINKED ISSUE #2135: [BUG] /me messages are sent to room and still kept in the input box
## Description

When sending `/me` messages via the android app, the text is sent to the room as `_message_` but its also kept in the input box after being sent.
On the same note, messages sent via `/me` are seen as `_message_` while those sent with `_message_` are properly displayed as _`message`_
## Devices and Versions

Your Rocket.Chat.Android version: 3.2.0
Your Rocket.Chat Server version: Official Rocket.chat server

Mobile device model and OS version: Huawei Y6 Pro 2017, Android 7.0

## Steps to reproduce
just type `/me message` and send it to the room. You'll get the described behaviour

### LINKED ISSUE #2190: Typing +:emoji_name: should react to last message
## Describe the feature you'd like

Typing +:emoji_name: should add a reaction to last message, just like in the desktop.

### PR DESCRIPTION
**Pull request by hfeky**
@RocketChat/android

Closes #2190
Closes #2135
Closes #2136

#### Changes:

1. When one types `+:emoji_name:` in the chat, it will automatically react to the last message, just as in the desktop version.
2. Fixed command text from not being cleared in the composer when entered.
3. Trimmed command params to fix `/me` command.

#### GIF for the change:

Emoji Reaction

### CODE REVIEW

File: app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt [hunk 1]
```diff
@@ -476,11 +480,17 @@ class ChatRoomFragment : Fragment(), ChatRoomView, EmojiKeyboardListener, EmojiR
...
         ui {
             if (!text.isBlank()) {
-                if (!text.startsWith("/")) {
-                    presenter.sendMessage(chatRoomId, text, editingMessageId)
-                } else {
+                if (text.startsWith("/")) {
                     presenter.runCommand(text, chatRoomId)
+                    return@ui
+                } else if (text.startsWith("+") && presenter.getLastMessageId() != null) {
+                    val trimmedText = text.substring(1).trimEnd()
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt]: Couldn't we move all these stuffs (trimming, etc) to the presenter?

File: app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt [hunk 2]
```diff
@@ -964,7 +973,7 @@ class ChatRoomFragment : Fragment(), ChatRoomView, EmojiKeyboardListener, EmojiR
         grantResults: IntArray
     ) {
         super.onRequestPermissionsResult(requestCode, permissions, grantResults)
-        when(requestCode) {
+        when (requestCode) {
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt:971]: Thanks!

Comment (hfeky) [reply app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt:971]: 😉

File: app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt (hunk 1 above)
Comment (hfeky) [reply app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt]: Sure, moved it. Please review changes.

**Review by philipbrito — APPROVED** [review]
Thank you very much for sending this PR @HusseinElFeky!
💯

**Pull request merged**